AK/Platform.h

```diff
@@ -29,6 +29,12 @@
 
 #define ARCH(arch) (defined(AK_ARCH_##arch) && AK_ARCH_##arch)
 
+#if !ARCH(I386) && !ARCH(X86_64)
```

Comment:
nit pick: As the reader, it always seems more pleasant to read code "straight threw" than to have to mentally parse negations. Consider inverting the ifdef: 

```cpp
#if ARCH(I386) || ARCH(X86_64)
#    define VALIDATE_IS_X86()
#else
#    define VALIDATE_IS_X86() static_assert(false, "Trying to include x86 only header on non x86 platform");
#endif
```